server/rooms.ts

```diff
@@ -671,7 +671,9 @@ export abstract class BasicRoom {
 		// `,` is a delimiter used by a lot of /commands
 		// `|` and `[` are delimiters used by the protocol
 		// `-` has special meaning in roomids
-		if (newTitle.includes(',') || newTitle.includes('|') || newTitle.includes('[') || newTitle.includes('-')) {
+		// groupchats use `[` so allow it to bypass this and run the checks in the commands themselves
+		if (this.persist &&
+			(newTitle.includes(',') || newTitle.includes('|') || newTitle.includes('[') || newTitle.includes('-'))) {
```

Comment:
This sorta subverts the thing I was going for earlier. I'll try to find the currently running version on Main for this.

Reply:
```ts
function(newTitle, newID) {
	if (!newID) newID = toID(newTitle);
	// `,` is a delimiter used by a lot of /commands
	// `|` and `[` are delimiters used by the protocol
	// `-` has special meaning in roomids
	if (newTitle.includes(',') || newTitle.includes('|')) {
		throw new Chat.ErrorMessage(`Room title "${newTitle}" can't contain any of: ,|`);
	}
	if ((!newID.includes('-') || newID.startsWith('groupchat-')) && newTitle.includes('-')) {
		throw new Chat.ErrorMessage(`Room title "${newTitle}" can't contain -`);
	}
	if (newID.length > 225) throw new Chat.ErrorMessage("The given room title is too long.");
	if (Rooms.search(newTitle)) throw new Chat.ErrorMessage(`The room '${newTitle}' already exists.`);
}
```

is what's currently running in Main.

Reply:
why not add that to the repo then while running it on main at the same time? main has a lot of forks and if any of them update the errors then affect them too.

Reply:
I was going to, but it was late and it needed to be cleaned up better. Feel free to add it to your PR, or I can just push it if you won't.

Reply:
okay, i just need 15 minutes then i'll edit it